docs/ko/docs/tutorial/sql-databases.md

```diff
@@ -0,0 +1,679 @@
+
+예를 들어, <a href="https://docs.celeryproject.org" class="external-link" target="_blank">Celery</a>, <a href="https://python-rq.org/" class="external-link" target="_blank">RQ</a>, 또는 <a href="https://arq-docs.helpmanual.io/" class="external-link" target="_blank">ARQ</a>와 함께 백그라운드 작업자에서의 사용이 가능합니다.
+
+## 모든 파일 검토
+
+`sql_app`을 하위 디렉토리로 갖는`my_super_project`디렉토리가 있어야 합니다.
+
+`sql_app`에는 다음의 파일들이 있습니다:
+
+- `sql_app/__init__.py`: 빈 파일
+
+- `sql_app/database.py`:
+
+```Python
+{!../../../docs_src/sql_databases/sql_app/database.py!}
+```
+
+* `sql_app/models.py`:
+
+```Python
+{!../../../docs_src/sql_databases/sql_app/models.py!}
+```
+
+* `sql_app/schemas.py`:
+
+```Python
+{!../../../docs_src/sql_databases/sql_app/schemas.py!}
+```
+
+* `sql_app/crud.py`:
+
+```Python
+{!../../../docs_src/sql_databases/sql_app/crud.py!}
+```
+
+* `sql_app/main.py`:
+
+```Python
+{!../../../docs_src/sql_databases/sql_app/main.py!}
+```
+
+## 확인하십시오
+
+이 코드를 복사해서 그대로 사용할 수 있습니다.
+
+!!! info "정보"
+
+    사실 이 문서들의 대부분의 코드와 마찬가지로 여기에 표시된 코드는 테스트의 일부입니다.
+
+Uvicorn을 통해 실행합니다:
+
+
+<div class="termy">
+
+```console
+$ uvicorn sql_app.main:app --reload
+
+<span style="color: green;">INFO</span>:     Uvicorn running on http://127.0.0.1:8000 (Press CTRL+C to quit)
+```
+
+</div>
+
+그다음, <a href="http://127.0.0.1:8000/docs" class="external-link" target="_blank">http://127.0.0.1:8000/docs</a>에서 브라우저를 열 수 있습니다.
+
+이제 FastAPI 응용 프로그램과 상호작용하며 실제 데이터베이스에서 데이터를 읽는 것이 가능합니다:
+
+<img src="/img/tutorial/sql-databases/image01.png">
+
+## 데이터베이스와 직접적인 상호작용
+
+FastAPI와 관계 없이 디버깅을 하거나, 테이블, 컬럼, 레코드를 추가하거나, 데이터를 수정하는 등의 작업을 위해  SQLite 데이터베이스(파일)를 직접 실행하고 싶다면, <a href="https://sqlitebrowser.org/" class="external-link" target="_blank">DB Browser for SQLite</a>를 사용할 수 있습니다.
+
+그것은 다음과 같습니다:
+
+<img src="/img/tutorial/sql-databases/image02.png">
+
+<a href="https://inloop.github.io/sqlite-viewer/" class="external-link" target="_blank">SQLite Viewer</a>나 <a href="https://extendsclass.com/sqlite-browser.html" class="external-link" target="_blank">ExtendsClass</a>와 같은 온라인 SQLite 브라우저를 사용할 수도 있습니다.
+
+## 미들웨어가 있는 대체 DB 세션
+
+**파이썬 3.7**을 사용하고 있지 않거나 **파이썬 3.6**에서 상기 언급한 “backports”를 설치할 수 없어 `yield`를 사용한 의존성을 사용할 수 없다면 유사한 방식으로 “미들웨어”에 세션을 설정할 수 있습니다.
+
+“미들웨어”는 요청이 있을 때마다 실행되는 함수로, 일부는 사전에, 일부는 함수의 종점 이후에 실행됩니다.
+
+### 미들웨어 생성
+
+우리가 추가할 미들웨어(단순한 함수)는 각 요청마다 새로운 SQLAlchemy `SessionLocal`을 생성, 요청에 이를 추가하고 요청이 완료되면 이를 종료할 것입니다.
+
+```Python hl_lines="14-22"
+{!../../../docs_src/sql_databases/sql_app/alt_main.py!}
+```
+
+!!! info "정보"  
+    `SessionLocal()` 의 생성과 요청의 처리를 `try` 블록에 배치합니다.
+
+    그리고 `finally` 블록에서 이를 종료합니다.
+
+    이를 통해 요청 이후 데이터베이스 세션이 언제나 닫혀있다는 사실을 확실히 할 수 있습니다. 요청 처리 도중 예외가 발생한 경우라도 그러합니다.
+
+### `request.state`에 대해
+
+`request.state`는 각 `Request` 객체에 대한 프로퍼티입니다. 이 경우에서의 데이터베이스 세션과 같이 요청 자체와 연결된 임의의 객체를 저장하기 위해 존재합니다. <a href="https://www.starlette.io/requests/#other-state" class="external-link" target="_blank">`Request` 상태에 관한 Starlette 문서</a>에서 더 자세한 사항을 확인할 수 있습니다.
```

Comment:
I think it's better to use "특성" or "재산" instead of "프로퍼티".

Reply:
Thank you for your feedback!🤗 However, I'm not sure whether it's a better idea, because I think it makes the meaning quite ambiguous and unclear. Plus, according to [this document](https://www.flowdas.com/pages/python-docs-ko.html), the official translation is "프로퍼티".

Reply:
It makes sense! thank you for giving your opinion! 😄